src/file/tokio_compat_file.rs

```diff
@@ -469,6 +474,12 @@ impl AsyncWrite for TokioCompatFile {
             .map(|n| min(n, max_write_len))
             .unwrap_or(max_write_len);
 
+        if self.write_len.fetch_add(n as usize, Ordering::Relaxed) + (n as usize)
+            > self.get_auxiliary().tokio_compat_file_write_limit()
+        {
+            ready!(self.as_mut().poll_flush(cx))?;
+        }
```

Comment:
After changing `write_len` from `AtomicUsize` => `usize`:

```suggestion
        let write_limit = self.get_auxiliary().tokio_compat_file_write_limit();
        let write_len = self.project().write_len;
        match write_len.checked_add(n as usize) {
            Some(new_write_len) if new_write_len > write_limit => {
                ready!(self.as_mut().poll_flush(cx))?;
            }
            None => {
                // overflow. This has to be a separate cases since
                // write_limit could be set to usize::MAX, in which case
                // saturating_add would never return anything larger
                // than it.
                ready!(self.as_mut().poll_flush(cx))?;
            }
            Some(new_write_len) => {
                *write_len = new_write_len;
            }
        }
```

also note that `AtomicUsize::fetch_add` could overflow (on 32-bit platform, extremely less likely on 64-bit platform but still possible), so it's better to use `saturating_add`.

I also noticed that `fetch_add` increments `write_len` even if no write is done.